apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -1747,7 +1747,21 @@ block_to_endpoint_map(Block) ->
 block_to_endpoint_map(Block, Options) ->
     Encoding = maps:get(tx_encoding, Options, message_pack),
     BMap = aec_blocks:serialize_client_readable(Encoding, Block),
-    aehttp_dispatch_ext:cleanup_genesis(BMap).
+    Expected = aehttp_dispatch_ext:cleanup_genesis(BMap),
+
+    %% Validate that all transactions have the correct block height and hash
+    ExpectedTxs = maps:get(<<"transactions">>, Expected, []),
```

Comment:
This default `[]` may cause things to slip though. Shall this rather case/switch on height of block, and if not genesis then hardcode expectation that `<<"transactions">>` is there otherwise hardcode expectation that `<<"transactions">>` is not there hence return `[]`?